src/sqs.rs

```diff
@@ -85,6 +89,53 @@ pub async fn send_messages_concurrently<Msg: serde::Serialize, St: Stream<Item =
         .await
 }
 
+/// The name of an AWS SQS queue.
+///
+/// The `FromStr`` implementation of this type ensures the value is a valid AWS
+/// SQS name. This means it is between 1 and 80 characters, and only contains
+/// alphanumberic characters, hyphens (-), and underscores (_).
+///
+/// Ref: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/quotas-queues.html
+#[derive(Clone, Debug, Display, From, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SQSQueueName(String);
+
+impl AsRef<str> for SQSQueueName {
+    fn as_ref(&self) -> &str {
+        &self.0
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum SQSQueueNameError {
+    #[error("Min length is 1")]
+    TooShort,
+    #[error("Max length is 80 characters")]
+    TooLong,
```

Comment:
Suggestion (non-blocking): We could reduce these 2 variants to one, whilst also returning the caller the actual received length, also helps at call-site as it gives users 1 variant to match against for "this" failure-case, rather than 2. For example:

```rust
#[error("Invalid length, expected between 1 and 80 characters, received: {0}")]
InvalidLength(usize),
...
```

Reply:
Yeah, that's actually nicer, let me update this 👍